ui/src/pages/plan/rules/ModifyRuleForm.tsx

```diff
@@ -56,37 +55,40 @@ export const ModifyForm = ({
     } else {
         ruleByMonthDay = 1;
     }
-
-    if (rruleObject.origOptions.bymonth != undefined) {
-        if (Array.isArray(rruleObject.origOptions.bymonth)) {
-            ruleByMonth = rruleObject.origOptions.bymonth[0];
+    
+    if (rruleObject.origOptions.byweekday != undefined) {
+        if (Array.isArray(rruleObject.origOptions.byweekday)) {
+            ruleWeekDays = rruleObject.origOptions.byweekday.map(wd => wd.toString());
         } else {
-            ruleByMonth = rruleObject.origOptions.bymonth;
+            ruleWeekDays = [rruleObject.origOptions.byweekday.toString()]
         }
-    } else {
-        ruleByMonth = 1;
-    }
-
-    if (rruleObject.origOptions.wkst != undefined) {
-        ruleWeekDays = [rruleObject.origOptions.wkst.toString()];
+        const rRuleWeekdayToWeekdayMapping: { [key: string]: string } = {
+            "SU": "SUNDAY",
+            "MO": "MONDAY",
+            "TU": "TUESDAY",
+            "WE": "WEDNESDAY",
+            "TH": "THURSDAY",
+            "FR": "FRIDAY",
+            "SA": "SATURDAY",
+        }
+        ruleWeekDays = ruleWeekDays.map(rruleWeekday => rRuleWeekdayToWeekdayMapping[rruleWeekday]);   
     }
 
     if (rruleObject.origOptions.dtstart != undefined){
         currentStartDate = rruleObject.origOptions.dtstart;
-        ruleStartDate = (currentStartDate.getFullYear().toString() + "-" + ((currentStartDate.getMonth() > 8) ? (currentStartDate.getMonth() + 1) : ('0' + (currentStartDate.getMonth() + 1)))).toString() + '-' + (((currentStartDate.getDate() > 9) ? currentStartDate.getDate() : ('0' + currentStartDate.getDate()))).toString();
+        ruleStartDate = (currentStartDate.getUTCFullYear().toString() + "-" + ((currentStartDate.getUTCMonth() > 8) ? (currentStartDate.getUTCMonth() + 1) : ('0' + (currentStartDate.getUTCMonth() + 1)))).toString() + '-' + (((currentStartDate.getUTCDate() > 9) ? currentStartDate.getUTCDate() : ('0' + currentStartDate.getUTCDate()))).toString();
     }
 
     if (rruleObject.origOptions.until != undefined) {
         currentEndDate = rruleObject.origOptions.until;
-        ruleEndDate = (currentEndDate.getFullYear().toString() + "-" + ((currentEndDate.getMonth() > 8) ? (currentEndDate.getMonth() + 1) : ('0' + (currentEndDate.getMonth() + 1)))).toString() + '-' + (((currentEndDate.getDate() > 9) ? currentEndDate.getDate() : ('0' + currentEndDate.getDate()))).toString();
+        ruleEndDate = (currentEndDate.getUTCFullYear().toString() + "-" + ((currentEndDate.getUTCMonth() > 8) ? (currentEndDate.getUTCMonth() + 1) : ('0' + (currentEndDate.getUTCMonth() + 1)))).toString() + '-' + (((currentEndDate.getUTCDate() > 9) ? currentEndDate.getUTCDate() : ('0' + currentEndDate.getUTCDate()))).toString();
```

Comment:
Because my timezone is behind GMT, this was shifting start/end day back 1 day everytime I edited. With these changes, no drift should happen.